myhomework2.php

```diff
@@ -0,0 +1,76 @@
+<?php
+
+
+// не звертайте на цю функцію уваги
+// вона потрібна для того щоб правильно зчитати вхідні дані
+function readHttpLikeInput()
+{
+    $f = fopen('php://stdin', 'r');
+    $store = "";
+    $toread = 0;
+
+    while ($line = fgets($f)) {
+        $store .= preg_replace("/\r/", "", $line);
+
+        if (preg_match('/Content-Length: (\d+)/', $line, $m))
+            $toread = $m[1] * 1;
+
+        if ($line === PHP_EOL)
+
+            break;
+    }
+
+    if ($toread > 0)
+        $store .= fread($f, $toread);
+
+    return $store;
+}
+
+$contents = readHttpLikeInput();
+
+function parseTcpStringAsHttpRequest($string)
+{
+
+    return [
+        "method" => getMethod($string),
```

Comment:
it's really cool that you made separate methods for getting method, uri, headers and body. Your approach is correct from a point of SOLID and architecture and so on.

But methods `getHeaders`, `getBody`, `getUri`, `getMethod` are doing explode of the same string for several times. Doing same work in code for several times is not efficient.
There is a way to a bit improve the logic to remove processing same data